dali/operators/reader/video_reader_decoder_op_test.cc

```diff
@@ -25,248 +25,239 @@
 
 
 namespace dali {
-class VideoReaderDecoderCpuTest : public VideoTestBase {
+class VideoReaderDecoderBaseTest : public VideoTestBase {
 };
 
-class VideoReaderDecoderGpuTest : public VideoTestBase {
-};
-
-TEST_F(VideoReaderDecoderCpuTest, CpuConstantFrameRate_CpuOnlyTests) {
-  const int batch_size = 4;
-  const int sequence_length = 6;
-  const int stride = 3;
-  const int step = 10;
-
-  Pipeline pipe(batch_size, 4, dali::CPU_ONLY_DEVICE_ID);
-
-  pipe.AddOperator(OpSpec("experimental__readers__Video")
-    .AddArg("device", "cpu")
-    .AddArg("sequence_length", sequence_length)
-    .AddArg("stride", stride)
-    .AddArg("step", step)
-    .AddArg(
-      "filenames",
-      std::vector<std::string>{
-        testing::dali_extra_path() + "/db/video/cfr/test_1.mp4",
-        testing::dali_extra_path() + "/db/video/cfr/test_2.mp4"})
-    .AddArg("labels", std::vector<int>{0, 1})
-    .AddOutput("frames", "cpu")
-    .AddOutput("labels", "cpu"));
-
-  pipe.Build({{"frames", "cpu"}, {"labels", "cpu"}});
-
-  int num_sequences = 20;
-  int sequence_id = 0;
-  int batch_id = 0;
-  int gt_frame_id = 0;
-
-  int video_idx = 0;
-
-  while (sequence_id < num_sequences) {
-    DeviceWorkspace ws;
-    pipe.RunCPU();
-    pipe.RunGPU();
-    pipe.Outputs(&ws);
-
-    auto &frame_video_output = ws.template Output<dali::CPUBackend>(0);
-    auto &frame_label_output = ws.template Output<dali::CPUBackend>(1);
-
-    for (int sample_id = 0; sample_id < batch_size; ++sample_id) {
-      const auto sample = frame_video_output.tensor<uint8_t>(sample_id);
-      const auto label = frame_label_output.tensor<int>(sample_id);
-
-      ASSERT_EQ(label[0], video_idx);
-
-      for (int i = 0; i < sequence_length; ++i) {
-        this->CompareFrames(
-          sample + i * this->FrameSize(video_idx),
-          this->GetCfrFrame(video_idx, gt_frame_id + i * stride),
-          this->FrameSize(video_idx));
-      }
+class VideoReaderDecoderCpuTest : public VideoReaderDecoderBaseTest {
+ public:
+  void RunTest(
+    std::vector<std::string> &videos_paths, std::vector<TestVideo> &ground_truth_videos) {
+    const int batch_size = 4;
+    const int sequence_length = 6;
+    const int stride = 3;
+    const int step = 10;
+
+    Pipeline pipe(batch_size, 4, dali::CPU_ONLY_DEVICE_ID);
+
+    pipe.AddOperator(OpSpec("experimental__readers__Video")
+      .AddArg("device", "cpu")
+      .AddArg("sequence_length", sequence_length)
+      .AddArg("stride", stride)
+      .AddArg("step", step)
+      .AddArg(
+        "filenames",
+        videos_paths)
+      .AddArg("labels", std::vector<int>{0, 1})
+      .AddOutput("frames", "cpu")
+      .AddOutput("labels", "cpu"));
+
+    pipe.Build({{"frames", "cpu"}, {"labels", "cpu"}});
+
+    int num_sequences = 20;
+    int sequence_id = 0;
+    int batch_id = 0;
+    int gt_frame_id = 0;
+
+    int video_idx = 0;
+
+    while (sequence_id < num_sequences) {
```

Comment:
```suggestion
    for (int sequence_id = 0; sequence_id < num_sequences; ++sequence_id) {
```
Reads better for me.